### PR TITLE
use Qt5/6 QMutex & QMutexLocker compatibility macros

### PR DESCRIPTION
**Pull request by Be-ing**

### CODE REVIEW

File: src/engine/engineworkerscheduler.cpp [hunk 1]
```diff
@@ -43,14 +44,14 @@ void EngineWorkerScheduler::run() {
     while (!m_bQuit) {
         Event::start(tag);
         {
-            QMutexLocker lock(&m_mutex);
+            const auto locker = lockMutex(&m_mutex);
```
Comment (uklotzde) [inline src/engine/engineworkerscheduler.cpp:47]: Why did you rename this occurrence manually?

For the future I suggest to use `const auto lockedMutex = lockMutex(...);` instead, but we don't need to do this in this PR.

Comment (Be-ing) [reply src/engine/engineworkerscheduler.cpp:47]: copying and pasting? :shrug:

Comment (uklotzde) [reply src/engine/engineworkerscheduler.cpp:47]: It looked like you used search&replace as a batch operation, but not for this occurrence. I was just wondering why.

**Review by uklotzde — APPROVED** [review]
Thank you! LGTM

**Pull request merged**